packages/commonwealth/client/scripts/views/pages/AdminPanel/UpdateCommunityIdTask.tsx

```diff
@@ -0,0 +1,113 @@
+    useState<boolean>(false);
+
+  const [newCommunityValue, setNewCommunityValue] = useState<string>('');
+  const [newValueValidated, setNewValueValidated] = useState<boolean>(false);
+
+  const openConfirmationModal = () => {
+    openConfirmation({
+      title: 'Update Community Id',
+      description:
+        `Are you sure you want to update ${originalCommunityValue} to ${newCommunityValue}? WARNING: this ` +
+        'does not set up redirects so this will break external links to the original community.',
+      buttons: [
+        {
+          label: 'Update',
+          buttonType: 'destructive',
+          buttonHeight: 'sm',
+          onClick: async () => {
+            try {
+              await updateCommunityId({
+                community_id: originalCommunityValue,
+                new_community_id: newCommunityValue,
+              });
+              setOriginalCommunityValue('');
+              setNewCommunityValue('');
+              notifySuccess('Community id updated');
+            } catch (e) {
+              notifyError('Error updating community id');
+              console.error(e);
+            }
+          },
+        },
+        {
+          label: 'Cancel',
+          buttonType: 'secondary',
+          buttonHeight: 'sm',
+        },
+      ],
+    });
+  };
+
+  function validateFn(
+    this: { new: boolean },
+    value: string,
+  ): [ValidationStatus, string] | [] {
+    const communityExists = app.config.chains.getById(value);
+
+    if (communityExists && this.new) {
+      setNewValueValidated(false);
+      return ['failure', 'Community already exists'];
+    } else if (!communityExists && !this.new) {
+      setOriginalValueValidated(false);
+      return ['failure', 'Community not found'];
+    }
+
+    if (this.new) setNewValueValidated(true);
+    else setOriginalValueValidated(true);
+    return [];
+  }
+
+  return (
+    <div className="TaskGroup">
+      <CWText type="h4">Update Community Id</CWText>
+      <CWText type="caption">
+        Updates a communities url e.g. commonwealth.im/cmn-protocol to
+        commonwealth.im/common. This does not update the Community name.
+        WARNING: This does not set up redirects so this will break external
+        links to the original community.
+      </CWText>
+      <div className="TaskRow">
+        <CWTextInput
+          value={originalCommunityValue}
+          onInput={(e) => {
+            setOriginalCommunityValue(e.target.value);
+            if (e.target.value.length === 0) setOriginalValueValidated(false);
+          }}
+          inputValidationFn={validateFn.bind({ new: false })}
+          placeholder="Enter a community id"
+        />
+        <CWTextInput
+          value={newCommunityValue}
+          onInput={(e) => {
+            setNewCommunityValue(e.target.value);
+            if (e.target.value.length === 0) setNewValueValidated(false);
+          }}
+          inputValidationFn={validateFn.bind({ new: true })}
```

Comment:
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/5150e8f5-c823-4b1d-ada3-297edc72309e)

error text is not visible, only red border

Reply:
we shuold use `slugifyPreserveDashes` . Otherwise, I can type something like `ade asdasdasd ` which results in community being accessible via `localhost:8080/ade%20asdasdasd` which is probably not desired